airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_table.py

```diff
@@ -171,6 +171,176 @@ def test_given_three_pages_read_then_return_records(self, uuid_mock, mock_auth,
         output = _read(_config(), sync_mode=SyncMode.full_refresh)
         assert len(output.records) == 3
 
+    
+
+    @parameterized.expand(
+            [
+                (["TEST_COLUMN_20"],12, 0,3 , "1970-01-04"),
+                (["TEST_COLUMN_21"], 13,30000, 300001, "1970-01-04T11:20:01.000000"),
+                (["TEST_COLUMN_2"], 11,3, 4, 4),
+                (["TEST_COLUMN_14"], 5,"aa", "ab", "ab")
+            ],
+    )
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+    @mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+    @HttpMocker()
+    def test_given_cursor_set_state(
+        self,
+        cursor_field,
+        cursor_index,
+        value_first_record,
+        value_second_record,
+        expected_state_cursor_value,
+        uuid_mock, 
+        mock_auth,
+        http_mocker: HttpMocker,
+        ) -> None:
+
+        def snowflake_record():
+            return a_snowflake_response("response_get_table",JsonPath("$.'data'[*]"), cursor_path=JsonPath(f"$.[{cursor_index}]"))
+
+        _given_get_timezone(http_mocker)
```

Comment:
nit: If those four are common to all tests, we could extract this into the `setUp` method

Reply:
> nit: If those four are common to all tests, we could extract this into the `setUp` method

I don't see how to do it because :

1. we need these four func and the test to be in the same context
2. we need to decorate the function in order to have the request matching
3. the is ne way to get the next test func called in the setUp function

if you have any idea i'd be curious to know , i may be missing something here.

Reply:
We have a couple of cases where we do that. For example: https://github.com/airbytehq/airbyte/blob/1375a433bbb28e7a00a0b05e332c2a643118388f/airbyte-integrations/connectors/source-salesforce/unit_tests/integration/test_bulk_stream.py#L61-L70

I think there is a known issue where the [exit statement](https://github.com/airbytehq/airbyte/blob/100f4e0f34e98dbc540790a9d836f511cfd4aa51/airbyte-cdk/python/airbyte_cdk/test/mock_http/mocker.py#L44-L45) does not validate the HttpRequest like [here](https://github.com/airbytehq/airbyte/blob/100f4e0f34e98dbc540790a9d836f511cfd4aa51/airbyte-cdk/python/airbyte_cdk/test/mock_http/mocker.py#L118-L124) but we are fine with this drawback given the readability it adds. I'll work on a fix on this today given I have some free time